app/models/user.rb

```diff
@@ -80,4 +80,12 @@ def requested?(story)
   def owns?(story)
     story.owned_by_id == id
   end
+
+  def team_from_project(project)
+    project
+      .teams
+      .map { |team| team if teams.include? team }
```

Comment:
Two tips here:

1 - This line makes it very implicit/hidden that the `teams` variable is coming from the model, I took some times to notice where `teams` was coming from and how this was working. We can improve that

2 - Usually, if you need to do a `map` + a `compact` it's a case where you should be using a [`select`](https://apidock.com/ruby/Array/select) :smile: Also, maybe we can filter this with a query instead of at the Ruby side? I don't know if it's gonna work but it would be something like this

```rb
user_team_ids = teams.pluck(:id)

project.teams.where(id: user_team_ids).first
```

If it doesn't work we try another approach at the Ruby side but using select instead of map + compact :smile: